sdk/azcore/arm/resource_type.go

```diff
@@ -0,0 +1,107 @@
+//go:build go1.16
+// +build go1.16
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package arm
+
+import (
+	"fmt"
+	"strings"
+)
+
+var (
+	// SubscriptionResourceType is the ResourceType of a subscription
+	SubscriptionResourceType = NewResourceType(builtInResourceNamespace, "subscriptions")
+	// ResourceGroupResourceType is the ResourceType of a resource group
+	ResourceGroupResourceType = NewResourceType(builtInResourceNamespace, "resourceGroups")
+	// TenantResourceType is the ResourceType of a tenant
+	TenantResourceType = NewResourceType(builtInResourceNamespace, "tenants")
+	// ProviderResourceType is the ResourceType of a provider
+	ProviderResourceType = NewResourceType(builtInResourceNamespace, "providers")
+)
+
+// ResourceType represents an Azure resource type, e.g. "Microsoft.Network/virtualNetworks/subnets"
+type ResourceType struct {
+	Namespace string
+	Type      string
+	Types     []string
+
+	stringValue string
+}
+
+func (t *ResourceType) LastType() string {
+	return t.Types[len(t.Types)-1]
+}
+
+func (t *ResourceType) String() string {
+	return t.stringValue
+}
+
+// IsParentOf returns true when the receiver is the parent resource type of the child.
+func (t *ResourceType) IsParentOf(child *ResourceType) bool {
+	if child == nil {
+		return false
+	}
+	if !strings.EqualFold(t.Namespace, child.Namespace) {
+		return false
+	}
+	var types = splitStringAndOmitEmpty(t.Type, "/")
```

Comment:
Is this splitting really necessary, couldn't you just do `strings.EqualFold(t.Type, child.Type)`?

Reply:
This is testing whether a type is parent of the other, therefore we do not test if they are equal. But we could test one to be prefix of the other if we do not use splitting.

Reply:
It turns out that we already split them while constructing them. Therefore I removed the split here, and directly use `child.Types`